c++/src/kj/compat/http.c++

```diff
@@ -6186,14 +6199,11 @@ private:
 
       auto pipe = newTwoWayPipe();
 
-      kj::Own<kj::AsyncIoStream> io =
-          kj::heap<DelayedEofIoStream>(kj::mv(pipe.ends[0]), task.attach(kj::addRef(*this)));
```

Comment:
What we really need here is for DelayedEofIoStream to inject the delay before the stream is ended in _both_ directions, that is, delay the second EOF that passes over it. However, there currently isn't really a way to do that, because sending an EOF is not an async operation... that's something I've been meaning to fix, by adding an `AsyncOutputStream::end()` method. Until then, I don't really know what to do here.

If you can avoid using the adapter entirely, that would probably be best for now.